bin/ch-pull2tar

```diff
@@ -20,12 +20,46 @@ EOF
 
 parse_basic_args "$@"
 
-if [ "$#" -ne 2 ]; then
+if [ "$#" -lt 2 ]; then
     usage
 fi
 
 image=$1
 dest=$2
+tar="$image".tar.gz
 
-docker_ pull "$image"
-"${ch_bin}/ch-builder2tar" "$image" "$dest"
+while [ $# -gt 0 ]; do
+    opt=$1; shift
+    case $opt in 
+        -p|--pull-from) 
+            if [ "$1" = "docker" ]; then
+                docker=yes
+            elif [ "$1" = "skopeo" ]; then 
+                skopeo=yes
+            else 
+                echo "$1 is an invalid option" 1>&2
+                exit 1
+            fi
+            ;;
+    esac
+done
+
+if [ -n "$docker" ] || ( command -v docker >/dev/null 2>&1 ); then
+    echo "Docker"
+    docker_ pull "$image"
+    # shellcheck disable=SC2154
+    "${ch_bin}/ch-builder2tar" -b docker "$image" "$dest"
+elif [ -n "$skopeo" ] || ( command -v skopeo >/dev/null 2>&1 ); then 
+    echo "Skopeo"
+    temp=$(mktemp -d --tmpdir ch-pull2tar.XXXXXX)
+    skopeo copy docker://"$image" oci:"$temp"/oci:"$image"
+    umoci unpack --rootless --image "$temp"/oci:"$image" "$temp"/img
+    ( cd "$temp"/img/rootfs/
+    # shellcheck disable=SC2035
+    tar czf ../"$tar" * )
```

Comment:
1. Put this subshell on a single line, for clarity.
2. Don't disable the ShellCheck warning; eliminate it. In this case, you don't need the glob. Instead, just tar `rootfs`; Charliecloud can deal with a top-level directory in the tarball.

Reply:
I chose this particular method because everything else was giving me the `implausibly old timestamp` error. I'll look for an alternative. I just tried ( cd $temp/img;  tar czf ../image.tar.gz rootfs/ ) and that resulted in:
```
[rustyd@mingus ch-pull2tar.GnLSWi]$ ch-tar2dir image.tar.gz .
creating new image ./image
tar: rootfs: implausibly old time stamp 1969-12-31 19:00:00
./image unpacked ok
```

Reply:
I think the solution is to find the directory that has the bad timestamp (`rootfs`?) and `touch(1)` it. We've seen this before; I think it's a bug in the base image.

Reply:
It ended up being a permissions issue all along.